ipaserver/dcerpc.py

```diff
@@ -162,11 +162,20 @@ class TrustTopologyConflictSolved(Exception):
     """
     pass
 
-def assess_dcerpc_exception(num=None, message=None):
+
+def assess_dcerpc_error(error):
     """
     Takes error returned by Samba bindings and converts it into
     an IPA error class.
     """
+    if isinstance(error, RuntimeError):
+        error_tuple = error.args
+    else:
+        error_tuple = error
+    if len(error_tuple) != 2:
+        raise RuntimeError("Unable to parse error: {err!r}".format(err=error))
+
```

Comment:
This is not a correct change because we used to call `assess_dcerpc_exception()` with either both `num` and `message` or with just a `num`. The change here assumes we always have a tuple or an error class which is not true.  One practical example is `verify_trust()` method: it calls `assess_dcerpc_exception()` with a single numeric return code`pdc_connection_status`.

Reply:
Judging from the `verify_trust()` code, I don't see where only `num` would be passed. `result.pdc_connection_status` has to have at least two members otherwise the check:
```python
if result.pdc_connection_status[0] != 0 and \
               result.tc_connection_status[0] != 0:
                if result.pdc_connection_status[1] == "WERR_ACCESS_DENIED":
```
would have thrown an exception. Is there a way only a single number would get there?

Reply:
I checked Samba code. `pdc_connection_status` is `WERROR`, the code is generated from an IDL definition and boils down to the following:
```
static PyObject *py_netr_NETLOGON_INFO_2_get_pdc_connection_status(PyObject *obj, void *closure)
{
        struct netr_NETLOGON_INFO_2 *object = (struct netr_NETLOGON_INFO_2 *)pytalloc_get_ptr(obj);
        PyObject *py_pdc_connection_status;
        py_pdc_connection_status = PyErr_FromWERROR(object->pdc_connection_status);
        return py_pdc_connection_status;
}
```
where `PyErr_FromWERROR()` is a macro that expands to

```
#define PyErr_FromWERROR(err) Py_BuildValue("(i,s)", W_ERROR_V(err), discard_const_p(char, win_errstr(err)))
```
so, yes, it expands to a tuple of an int and string. Good.

Reply:
Thank you for the thorough research :)